src/satosa/frontends/base.py

```diff
@@ -14,17 +17,22 @@ def __init__(self, auth_req_callback_func, internal_attributes, base_url, name):
         :type auth_req_callback_func:
         (satosa.context.Context, satosa.internal.InternalData) -> satosa.response.Response
         :type internal_attributes: dict[str, dict[str, str | list[str]]]
+        :type base_url: str
         :type name: str
 
         :param auth_req_callback_func: Callback should be called by the module after the
         authorization response has been processed.
+        :param internal_attributes: attribute mapping
+        :param base_url: base url of the proxy
         :param name: name of the plugin
         """
         self.auth_req_callback_func = auth_req_callback_func
         self.internal_attributes = internal_attributes
         self.converter = AttributeMapper(internal_attributes)
-        self.base_url = base_url
+        self.base_url = base_url or ""
         self.name = name
+        self.endpoint_baseurl = os.path.join(self.base_url, self.name)
```

Comment:
`os.path.join` will not work on certain platforms (see Windows..).

To join URLs with paths using a function, use `urllib.parse.urljoin`; but it also has caveats (paths that begin with `/` will be considered the root, and bases that do not end with a `/` will be considered a file and will be truncated).

The simplest approach is to just concatenate with a `/` (ie, `f"{self.base_url}/{self.name}"`).

Reply:
I would probably name the function `join_url_paths` (verb first).

Note that the latest pysaml2 already requires Python 3.9 and SATOSA will be updated to require it too. IdentityPython projects try to be compatible with the python that ships on the latest Debian stable release (which is now Python 3.9).
So, requiring Python 3.9 is fine; but no newer atm.

Reply:
The updated patch contains a `join_paths` implementation, which tries to handle separators a little bit more intelligently than a simple concatenation. I've replaced all erroneous `os.path.join` calls to `join_paths`, but didn't replace `{}/{}.format(foo, bar)` all over the code, since this appears way too many times.